src/core/store/webstorage.js

```diff
@@ -0,0 +1,336 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2016-2019 (original work) Open Assessment Technologies SA ;
+ */
+
+/**
+ * WebStorage backend of the client store
+ *
+ * @author Bertrand Chevrier <bertrand@taotesting.com>
+ */
+
+import _ from 'lodash';
+import promiseQueue from 'core/promiseQueue';
+import uuid from 'lib/uuid';
+
+/**
+ * Prefix all databases
+ * @type {String}
+ */
+var prefix = 'tao-store-';
+
+/**
+ * The name of the store that contains the index of known stores.
+ * @type {String}
+ */
+var knownStoresName = 'index';
+
+/**
+ * The name of the store that contains the store id
+ * @type {String}
+ */
+var idStoreName = 'id';
+
+var writingQueue = promiseQueue();
```

Comment:
Caution, this variable is scoped to the whole module, and therefore will be shared with every instance of the `webStorage`. That means the same queue will be targetted for `localStorage` and for `sessionStorage`. Is it intended? Maybe would it be better to move it inside the factory instead.

Reply:
You are right. I will move it.